bot/exts/moderation/infraction/infractions.py

```diff
@@ -351,9 +355,13 @@ async def apply_voice_ban(self, ctx: Context, user: UserSnowflake, reason: t.Opt
         if reason:
             reason = textwrap.shorten(reason, width=512, placeholder="...")
 
-        await user.move_to(None, reason="Disconnected from voice to apply voiceban.")
+        action = None
```

Comment:
Something strange I noticed is that the infraction won't be scheduled to expire if it doesn't have an `action`.

https://github.com/python-discord/bot/blob/41e861c52c5e8c3f5fb751345661d8f062976d87/bot/exts/moderation/infraction/_scheduler.py#L176-L182

I'm not sure why it was done this way, but it means that a temp voice ban will only expire when the bot restarts and infractions are rescheduled. A mute doesn't suffer from this since the type check is inside the action.

Reply:
From what I gather, the reason it was done like this is so you can have non-expiring infractions (notes and warnings for example). It is probably unintuitive behavior, and it is probably better for the scheduling to be an optional argument.

Either way, I'll add a note in the docstring for now, but I can implement my suggestion if you believe it'll be more useful.

Reply:
I don't think that's the reason. It already uses `if expiry`, so setting `expiry` to `None` is a way to not schedule expiration. The confusing part is why it's _also_ dependent on an action being given.